app.rb

```diff
@@ -0,0 +1,171 @@
+      case selected_option
+      when 1 then list_all_books
+      when 2 then list_all_people
+      when 3 then create_teacher_or_student
+      when 4 then create_a_book
+      when 5 then create_rental
+      when 6 then list_rentals_for_a_person
+      when 7
+        puts 'Thanks for using this app'
+        exit
+      else puts 'Error: Enter a valid option.'
+      end
+    end
+  end
+
+  # rubocop:enable Metrics/CyclomaticComplexity
+  private
+
+  def homepage
+    puts 'Welcome to school library App'
+    puts 'Please choose an option by entering in a number'
+    puts '1. List all books'
+    puts '2. List all people'
+    puts '3. Create a person(teacher or student, not a plain person)'
+    puts '4. Create a book'
+    puts '5. Create rental'
+    puts '6. List rentals for a given person id'
+    puts "7. Quit\n"
+  end
+
+  def list_all_books
+    puts 'List of all books'
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}\n"
+      puts
+    end
+  end
+
+  def list_all_people
+    puts 'List of all people'
+    @people.each do |person|
+      puts "[#{person.class}] name: #{person.name}, ID: #{person.id}, age: #{person.age}\n"
+      puts
+    end
+  end
+
+  def create_teacher_or_student
+    puts 'Press 1 to create a Teacher or 2 to create a Student'
+    number_entered = gets.chomp.to_i
+    case number_entered
+    when 1 then create_a_teacher
+    when 2 then create_a_student
+    else
+      puts 'You can only create a teacher(1) or student(2)'
+    end
+    # puts
+  end
+
+  def create_a_teacher
+    puts 'Creating a new teacher...'
+    puts "Enter the teacher's name:"
+    name = gets.chomp
+    puts "Enter the teacher's age:"
+    age = gets.chomp.to_i
+    puts "Enter the teacher's specialization:"
+    specialization = gets.chomp
+
+    teacher = Teacher.new(age, name, specialization)
+    @people << teacher
+    puts "#{teacher.name} who specializes in #{teacher.specialization} has been created successfully.\n"
+    # puts
+  end
+
+  def create_a_student
+    puts 'Creating a new student...'
+    puts "Enter the student's name:"
+    name = gets.chomp
+    puts "Enter the student's age:"
+    age = gets.chomp.to_i
+    puts "Do the student have parent's permission? (y/n)"
+    parent_permission = gets.chomp.downcase == 'y'
+
+    student = Student.new(age, name, parent_permission)
+    @people << student
+    puts "#{student.name} has been created successfully.\n"
+  end
+
+  def create_a_book
+    puts 'Creating a new book...'
+    puts "Enter the book's title:"
+    title = gets.chomp
+    puts "Enter the book's author:"
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    @books << book
+    puts "#{book.title} has been created successfully.\n"
+    puts
+  end
+
+  def create_rental
+    book = select_book
+    person = select_person
+    rental_date = enter_rental_date
+    create_rental_record(rental_date, book, person)
+  end
```

Comment:
- [ ] Please your app can not create a rental so make sure you fix this error :point_down: to get this method of creating a rental to work perfectly. :pray: 
  
![image](https://user-images.githubusercontent.com/92303891/233313011-a5b49b51-0cf3-4a2f-b637-b739ef88329a.png)